frontend/tests/api/SearchOpportunityApi.test.ts

```diff
@@ -1,113 +1,219 @@
-import SearchOpportunityAPI from "src/app/api/SearchOpportunityAPI";
+import SearchOpportunityAPI, {
+  buildFilters,
+  buildPagination,
+} from "src/app/api/SearchOpportunityAPI";
 import { QueryParamData } from "src/services/search/searchfetcher/SearchFetcher";
-import { SearchRequestBody } from "src/types/search/searchRequestTypes";
-
-// mockFetch should match the SearchAPIResponse type structure
-const mockFetch = ({
-  response = {
-    data: [],
-    message: "Success",
-    pagination_info: {
-      order_by: "opportunity_id",
-      page_offset: 1,
-      page_size: 25,
-      sort_direction: "ascending",
-      total_pages: 1,
-      total_records: 0,
-    },
-    status_code: 200,
-    errors: [],
-    warnings: [],
+import { SearchFetcherActionType } from "src/types/search/searchRequestTypes";
+
+const validMockedResponse = {
+  data: [],
+  message: "Success",
+  pagination_info: {
+    // TODO: the response order_by should
+    // by what the request had: opportunity_number
+    order_by: "opportunity_id",
+    page_offset: 1,
+    page_size: 25,
+    sort_direction: "ascending",
+    total_pages: 1,
+    total_records: 0,
   },
-  ok = true,
-  status = 200,
-}) => {
-  return jest.fn().mockResolvedValueOnce({
-    json: jest.fn().mockResolvedValueOnce(response),
-    ok,
-    status,
-  });
+  status_code: 200,
+  warnings: [],
 };
 
-describe("SearchOpportunityAPI", () => {
-  let searchApi: SearchOpportunityAPI;
-  const baseRequestHeaders = {
-    "Content-Type": "application/json",
-  };
+const searchProps: QueryParamData = {
+  page: 1,
+  status: new Set(["forecasted", "posted"]),
+  fundingInstrument: new Set(["grant", "cooperative_agreement"]),
+  agency: new Set(),
+  category: new Set(),
+  eligibility: new Set(),
+  query: "research",
+  sortby: "opportunityNumberAsc",
+};
 
+let searchApi: SearchOpportunityAPI;
+const mockedRequest = jest.fn();
+
+describe("SearchOpportunityAPI", () => {
   beforeEach(() => {
+    searchApi = new SearchOpportunityAPI();
+    jest.spyOn(searchApi, "request").mockImplementation(mockedRequest);
+  });
+  afterEach(() => {
     jest.resetAllMocks();
+  });
 
-    searchApi = new SearchOpportunityAPI();
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("instantiates correctly", () => {
+    expect(searchApi.namespace).toEqual("opportunities");
   });
 
   describe("searchOpportunities", () => {
-    beforeEach(() => {
-      global.fetch = mockFetch({});
```

Comment:
this test was mocking down to the fetch level, while the existing opportunity listing API test was just mocking the request function on the class. In the interest of using inheritance to our benefit and keeping the tests somewhat uniform, I changed this to use the request mocking approach, and left the fetch mocking approach for the base class